### ISSUE TITLE
Makefile uses "uname -p" instead of "uname -m"

### ISSUE THREAD

**Issue by cduchesne** [emccode/rexray#437]
```
ifeq ($(origin GOARCH), undefined)
		UNAME_P := $(shell uname -p)
		ifeq ($(UNAME_P),x86_64)
			export GOARCH = amd64
		endif
		ifneq ($(filter %86,$(UNAME_P)),)
			export GOARCH = 386
		endif
	endif
```

The correct use is "uname -m" to get x86_64. -p will show unknown or i386 depending on operating system.

```
$ uname --help
Usage: uname [OPTION]...
Print certain system information.  With no OPTION, same as -s.

  -a, --all                print all information, in the following order,
                             except omit -p and -i if unknown:
  -s, --kernel-name        print the kernel name
  -n, --nodename           print the network node hostname
  -r, --kernel-release     print the kernel release
  -v, --kernel-version     print the kernel version
  -m, --machine            print the machine hardware name
  -p, --processor          print the processor type (non-portable)
  -i, --hardware-platform  print the hardware platform (non-portable)
  -o, --operating-system   print the operating system
      --help     display this help and exit
      --version  output version information and exit

```

**Comment by cduchesne** [comment]
modified Makefile with pull request #444

**Comment by cduchesne** [comment]
merged into milestone 0.4.0
